web/src/hooks/useSearch.ts

```diff
@@ -223,7 +224,7 @@ export const runSearch = async (
     const history = entries
       .filter((entry) => entry.role !== "error")
       .map((entry) => ({
-        role: entry.role,
+        role: (entry.deleted ? "deleted" : entry.role) as EntryRole,
```

Comment:
is there a reason to recycle `role` and not add a field `deleted`?

Reply:
not really. I initially added an extra field, but this has the nice property that it will raise an error if not handled properly, as langchain won't accept an unknown role